includes/discovery/sensors/snr/weos.inc.php

```diff
@@ -0,0 +1,31 @@
+<?php
+/*
+ * LibreNMS
+ *
+ * weos.inc.php
+ *
+ * LibreNMS snr discovery module module for Westermo WeOS
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * LibreNMS
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+foreach ($pre_cache['weos_hdsl2ShdslEndpointCurrTable'] as $index => $data) {
+    if (is_numeric($data['hdsl2ShdslEndpointCurrSnrMgn'])) {
+        // $index is 4096.0.0.1 but ifName is only 4096. Use explode to break the string into an array.
```

Comment:
I see 2 variants here : 
- keep the sensor as you did it, but write it in YAML.
- insert this hdsl/shdsl data in the xdsl discovery and poller module. Won't fit exactly the current sensors (they are made for ADSL and VDSL).

Reply:
I struggled with this one a lot. I tried writing it in YAML at first, but I couldn't get it to work properly.
I managed to get LibreNMS discover the SNR-values correctly when running discovery.php, but it didn't poll them correctly. When polling, it returned the value of the first port (1/1) for both ports (1/1 and 1/2). I couldn't get it to poll both ports separately.

Reply:
I have tried to write the SNR sensor discovery in YAML instead, and commented out the contents in the pre-cache files. I'll remove the pre-cache files completely if we get the YAML configuration to work.
However, I'm not able to get it to work properly with the sensor written in YAML.

I have attached a log with output from `discovery.php -d -m ports,sensors`.
I have also attached a screenshot from the LibreNMS webui for the device.

At the moment I see two problems:
1. The SNR value is not retrieved/shown correctly in LibreNMS.
2. LibreNMS is not able to replace "descr" with `Port {{ $ifName }} SNR`. Errors are shown in the discovery.php output.


The SNR-value for the SHDSL-ports are in the following OIDs:
Port 1/1 (ifIndex 4096): 1.3.6.1.2.1.10.48.1.5.1.2.4096.0.1.1
Port 1/2 (ifIndex 4097): 1.3.6.1.2.1.10.48.1.5.1.2.4097.1.1.1

```
[librenms]$ snmpwalk -v2c -c public -M /opt/librenms/mibs -m HDSL2-SHDSL-LINE-MIB <HOSTNAME> hdsl2ShdslEndpointCurrSnrMgn 
HDSL2-SHDSL-LINE-MIB::hdsl2ShdslEndpointCurrSnrMgn.4096.0.networkSide.wirePair1 = INTEGER: 27 dB
HDSL2-SHDSL-LINE-MIB::hdsl2ShdslEndpointCurrSnrMgn.4097.xtuC.networkSide.wirePair1 = INTEGER: 26 dB

[librenms]$ snmpwalk -v2c -c public -M /opt/librenms/mibs -m HDSL2-SHDSL-LINE-MIB <HOSTNAME> hdsl2ShdslEndpointCurrSnrMgn -On
.1.3.6.1.2.1.10.48.1.5.1.2.4096.0.1.1 = INTEGER: 27 dB
.1.3.6.1.2.1.10.48.1.5.1.2.4097.1.1.1 = INTEGER: 26 dB

[librenms]$ snmpwalk -v2c -c public -M /opt/librenms/mibs -m HDSL2-SHDSL-LINE-MIB <HOSTNAME>
[discovery debug output.txt](https://github.com/librenms/librenms/files/14066314/discovery.debug.output.txt)
![LibreNMS Westermo DDW-225 screenshot](https://github.com/librenms/librenms/assets/86343336/0af94fbb-c8a0-4351-9188-d9553a149156)
 ifName
IF-MIB::ifName.1 = STRING: lo
IF-MIB::ifName.5 = STRING: vlan1
IF-MIB::ifName.6 = STRING: vlan42
IF-MIB::ifName.7 = STRING: vlan71
IF-MIB::ifName.4096 = STRING: 1/1
IF-MIB::ifName.4097 = STRING: 1/2
IF-MIB::ifName.4098 = STRING: 2/1
IF-MIB::ifName.4099 = STRING: 2/2
IF-MIB::ifName.4100 = STRING: 2/3
IF-MIB::ifName.4101 = STRING: 2/4
```

[discovery debug output.txt](https://github.com/librenms/librenms/files/14066319/discovery.debug.output.txt)
![LibreNMS Westermo DDW-225 screenshot](https://github.com/librenms/librenms/assets/86343336/6a98cca1-6ab7-4c84-9a44-30365b3ee633)